Model/SourcesRepository.php

```diff
@@ -0,0 +1,178 @@
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterface;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionItemInterfaceFactory;
+
+/**
+ * Class SourcesRepository
+ * @package Ampersand\DisableStockReservation\Model
+ */
+class SourcesRepository
+{
+    /**
+     * @var SourcesFactory
+     */
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * @var \Magento\Framework\Api\DataObjectHelper
+     */
+    protected $dataObjectHelper;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultFactory;
+
+    /**
+     * @var SourceSelectionItemInterfaceFactory
+     */
+    private $sourceSelectionItemInterface;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     * @param SerializerInterface $serializer
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory
+     * @param SourceSelectionItemInterfaceFactory $sourceSelectionItemInterface
+     */
+    public function __construct(
+        SourcesFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SerializerInterface $serializer,
+        \Magento\Framework\Api\DataObjectHelper $dataObjectHelper,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory,
+        SourceSelectionItemInterfaceFactory $sourceSelectionItemInterface
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->serializer = $serializer;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->sourceSelectionResultFactory = $sourceSelectionResultFactory;
+        $this->sourceSelectionItemInterface = $sourceSelectionItemInterface;
+
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws NoSuchEntityException
+     */
+    public function getByOrderId(string $orderId): SourceModel
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        return $sourcesModel;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceSelectionResultInterface
+     */
+    public function getSourceSelectionResultByOrderId(string $orderId): SourceSelectionResultInterface
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        $sourcesArray = $this->serializer->unserialize($sourcesModel->getSources());
```

Comment:
Yeah I definitely think this logic should live in classes outside of this repository, maybe we need "converter" type functionality?